### ISSUE TITLE
Case-sensitive issue with sources/modifiers

### ISSUE THREAD

**Issue by snebjorn** [dotnet/templating#811]
This is probably related to #398 

If the condition compares a string, then only lower case strings will work.

The below will fail.
```
  "sources": [
    {
      "modifiers": [
        {
          "condition": "(Framework==\"Core\")",
          "exclude": ["TemplateName.Web/web.config"]
        },
        {
          "condition": "(Framework==\"Full\")",
          "exclude": ["TemplateName.Web/app.config"]
        }
      ]
    }
  ],
  "symbols": {
    "Framework": {
      "type": "parameter",
      "datatype": "choice",
      "choices": [
        {
          "choice": "Core",
          "description": "Target netstandard1.6"
        },
        {
          "choice": "Full",
          "description": "Target net461"
        }
      ],
      "defaultValue": "Core"
    }
  }
```
Lower casing Core and Full works. Is this intended?

**Comment by mlorbetske** [comment]
@seancpeters any ideas?

**Comment by seancpeters** [comment]
Strange, I'll take a look.

**Comment by seancpeters** [comment]
@snebjorn,
I'm unable to reproduce the behavior described. Can you copy-paste an entire template.json that is configured in a way that causes the problem, and the exact command you're running to instantiate the template? If you have a dotnetcli.host.json file for the template, please paste its contents here also.

Thanks!

**Comment by snebjorn** [comment]
Sure.

This is the troublesome part: (the entire thing is further down)
```
{
  "sources": [
    {
      "modifiers": [
        {
          "condition": "(SeedMethod != \"OOSeed\")",
          "exclude": [
            "src/Company.WebApplication1.Infrastructure.DataAccess/Data/Seed/**"
          ]
        },
        {
          "condition": "(SeedMethod != \"CSVSeed\")",
          "exclude": [
            "src/Company.WebApplication1.Infrastructure.DataAccess/CsvSeeder/**"
          ]
        }
      ]
    }
  ],
  "symbols": {
    "SeedMethod": {
      "type": "parameter",
      "datatype": "choice",
      "choices": [
        {
          "choice": "OOSeed",
          "description": "Object Orienated seed method"
        },
        {
          "choice": "CSVSeed",
          "description": "seed from a csv file"
        }
      ],
      "description": "Method for seeding the database with data.",
      "defaultValue": "OOSeed"
    }
  }
}
```
When I run `dotnet new3 mytemplate foofolder` both paths are excluded.

Now that I think about it. I might be running an older version. I can see my clone is 72 behind the master. I'll try to reinstall and see if that helps.

template.json
```
{
  "author": "Microsoft",
  "classifications": [
    "Web",
    "MVC"
  ],
  "name": "ASP.NET Core Web App",
  "generatorVersions": "[1.0.0.0-*)",
  "groupIdentity": "ITMinds.Web.Mvc",
  "precedence": "100",
  "identity": "ITMinds.Web.Mvc.CSharp",
  "shortName": "itm-mvc",
  "thirdPartyNotices": "https://aka.ms/template-3pn",
  "tags": {
    "language": "C#",
    "type": "project"
  },
  "sourceName": "Company.WebApplication1",
  "preferNameDirectory": true,
  "guids": [
    "53bc9b9d-9d6a-45d4-8429-2a2761773502", // Usersecrets
    "84614420-CD84-4090-8C5C-AE40EE5B9828", // src folder
    "9A710AD3-0898-4540-A98D-CF4C44923102", // MVC project
    "8BB5AA5E-CD33-4314-A910-9384285B9577", // Command project
    "46846053-0122-4F8C-999A-FBC83B752DF7", // DataAccess project
    "DED1A414-3FD0-4F10-A0C9-036DC1B3E601", // Entities project
    "CCB48643-5EE0-4C46-BE65-439982C1684F", // Query project
    "047416CD-0D54-4C35-8FE7-F9C93C122D16", // DomainServices project
    "662BA7AA-DE8A-46AF-A00B-90C78782F965", // test folder
    "429C736D-F8D0-4EB4-9FA2-EE1072638309", // Integration.Tests project
    "3BD99B54-EFF1-4085-B3B0-0F55AD3AD5DF", // MVC.Tests project
    "449E76AA-EE3D-402A-B609-30DE67B073BE", // Command.Tests project
    "07AB1420-9D98-443C-AFF8-75AAE82E785F", // Query.Tests project
    "0C5C9445-17FC-4EAA-8FEC-3165134B1803" // DataAccess.Tests project
  ],
  "sources": [
    {
      "exclude": [
        ".vs/**/*",
        ".vscode/**/*",
        "**/[Bb]in/**",
        "**/[Oo]bj/**",
        ".template.config/**/*",
        "**/*.filelist",
        "**/*.user",
        "**/*.lock.json"
      ],
      "modifiers": [
        {
          "condition": "(!IndividualAuth && !OrganizationalAuth)",
          "exclude": [
            "src/Company.WebApplication1.Application.MVC/Controllers/AccountController.cs",
            "src/Company.WebApplication1.Application.MVC/Views/Account/**",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/**",
            "src/Company.WebApplication1.Application.MVC/Views/Shared/_LoginPartial.cshtml"
          ]
        },
        {
          "condition": "(UseLocalDB || !IndividualAuth)",
          "exclude": [
            "*.db"
          ]
        },
        {
          "condition": "(!IndividualAuth)",
          "exclude": [
            "src/Company.WebApplication1.Application.MVC/Controllers/ManageController.cs",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ConfirmEmail.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ExternalLoginConfirmation.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ExternalLoginFailure.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ForgotPassword.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ForgotPasswordConfirmation.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/Lockout.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/Login.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/Register.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ResetPassword.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/ResetPasswordConfirmation.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/SendCode.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Account/VerifyCode.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/AddPhoneNumber.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/ChangePassword.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/Index.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/ManageLogins.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/SetPassword.cshtml",
            "src/Company.WebApplication1.Application.MVC/Views/Manage/VerifyPhoneNumber.cshtml",
            "src/Company.WebApplication1.Application.MVC/Services/**",
            "src/Company.WebApplication1.Application.MVC/Data/**",
            "src/Company.WebApplication1.Application.MVC/Models/**",
            "test/Company.WebApplication1.Application.MVC.Tests/Controllers/ManageControllerTests.cs"
          ]
        },
        {
          "condition": "(!IncludeLaunchSettings)",
          "exclude": [
            "src/Company.WebApplication1.Application.MVC/Properties/launchSettings.json"
          ]
        },
        {
          "condition": "(IndividualAuth)",
          "exclude": [
            "src/Company.WebApplication1.Application.MVC/Views/Account/SignedOut.cshtml"
          ]
        },
        {
          "condition": "(SeedMethod != \"OOSeed\")",
          "exclude": [
            "src/Company.WebApplication1.Infrastructure.DataAccess/Data/Seed/**"
          ]
        },
        {
          "condition": "(SeedMethod != \"CSVSeed\")",
          "exclude": [
            "src/Company.WebApplication1.Infrastructure.DataAccess/CsvSeeder/**"
          ]
        }
      ]
    }
  ],
  "symbols": {
    "auth": {
      "type": "parameter",
      "datatype": "choice",
      "choices": [
        {
          "choice": "None",
          "description": "No authentication"
        },
        {
          "choice": "Individual",
          "description": "Individual authentication"
        },
        {
          "choice": "SingleOrg",
          "description": "Organizational authentication for a single tenant"
        },
        {
          "choice": "MultiOrg",
          "description": "Organizational authentication for multiple tenants"
        },
        {
          "choice": "Windows",
          "description": "Windows authentication"
        }
      ],
      "defaultValue": "Individual",
      "description": "The type of authentication to use"
    },
    "AADInstance": {
      "type": "parameter",
      "datatype": "string",
      "defaultValue": "https://login.microsoftonline.com/",
      "replaces": "https:////login.microsoftonline.com/",
      "description": "The Azure Active Directory instance to connect to (use with Single-org or Multi-org auth types)."
    },
    "ClientId": {
      "type": "parameter",
      "datatype": "string",
      "replaces": "11111111-1111-1111-11111111111111111",
      "description": "The Client ID for this project (use with Single-org or Multi-org auth types)."
    },
    "Domain": {
      "type": "parameter",
      "datatype": "string",
      "replaces": "qualified.domain.name",
      "description": "The domain owning the directory (use with Single-org auth)."
    },
    "TenantId": {
      "type": "parameter",
      "datatype": "string",
      "replaces": "22222222-2222-2222-2222-222222222222",
      "description": "The TenantId ID of the directory to connect to (use with Single-org auth)."
    },
    "OrgReadAccess": {
      "type": "parameter",
      "datatype": "bool",
      "defaultValue": "false",
      "description": "Whether or not to allow this application read access to the directory (only applies to SingleOrg or MultiOrg auth types)."
    },
    "UserSecretsId": {
      "type": "parameter",
      "datatype": "string",
      "replaces": "aspnet-Company.WebApplication1-53bc9b9d-9d6a-45d4-8429-2a2761773502",
      "defaultValue": "aspnet-Company.WebApplication1-53bc9b9d-9d6a-45d4-8429-2a2761773502",
      "description": "The ID to use for secrets (use with OrgReadAccess or Individual auth)."
    },
    "IncludeLaunchSettings": {
      "type": "parameter",
      "datatype": "bool",
      "defaultValue": "false",
      "description": "Whether to include launchSettings.json in the generated template."
    },
    "HttpsPort": {
      "type": "parameter",
      "datatype": "integer",
      "description": "Port number to use to configure SSL in launchSettings.json."
    },
    "HttpsPortGenerated": {
      "type": "generated",
      "generator": "port"
    },
    "HttpsPortReplacer": {
      "type": "generated",
      "generator": "coalesce",
      "parameters": {
        "sourceVariableName": "HttpsPort",
        "fallbackVariableName": "HttpsPortGenerated"
      },
      "replaces": "43434"
    },
    "KestrelPort": {
      "type": "parameter",
      "datatype": "integer",
      "description": "Port number to use to configure Kestrel in launchSettings.json."
    },
    "KestrelPortGenerated": {
      "type": "generated",
      "generator": "port"
    },
    "KestrelPortReplacer": {
      "type": "generated",
      "generator": "coalesce",
      "parameters": {
        "sourceVariableName": "KestrelPort",
        "fallbackVariableName": "KestrelPortGenerated"
      },
      "replaces": "5000"
    },
    "IISExpressPort": {
      "type": "parameter",
      "datatype": "integer",
      "description": "Port number to use to configure IIS Express in launchSettings.json."
    },
    "IISExpressPortGenerated": {
      "type": "generated",
      "generator": "port"
    },
    "IISExpressPortReplacer": {
      "type": "generated",
      "generator": "coalesce",
      "parameters": {
        "sourceVariableName": "IISExpressPort",
        "fallbackVariableName": "IISExpressPortGenerated"
      },
      "replaces": "55555"
    },
    "OrganizationalAuth": {
      "type": "computed",
      "value": "(auth == \"SingleOrg\" || auth == \"MultiOrg\")"
    },
    "WindowsAuth": {
      "type": "computed",
      "value": "(auth == \"Windows\")"
    },
    "MultiOrgAuth": {
      "type": "computed",
      "value": "(auth == \"MultiOrg\")"
    },
    "SingleOrgAuth": {
      "type": "computed",
      "value": "(auth == \"SingleOrg\")"
    },
    "IndividualAuth": {
      "type": "computed",
      "value": "(auth == \"Individual\")"
    },
    "NoAuth": {
      "type": "computed",
      "value": "(!(IndividualAuth || MultiOrgAuth || SingleOrgAuth || WindowsAuth))"
    },
    "RequiresHttps": {
      "type": "computed",
      "value": "(OrganizationalAuth)"
    },
    "UseLocalDB": {
      "type": "parameter",
      "datatype": "bool",
      "defaultValue": "false",
      "description": "Whether or not to use LocalDB instead of SQLite"
    },
    "IncludeApplicationInsights": {
      "type": "parameter",
      "datatype": "bool",
      "defaultValue": "false",
      "description": "Whether or not to include Application Insights in the project"
    },
    "TargetFrameworkOverride": {
      "type": "parameter",
      "description": "Overrides the target framework",
      "replaces": "TargetFrameworkOverride",
      "datatype": "string",
      "defaultValue": ""
    },
    "Framework": {
      "type": "parameter",
      "description": "The target framework for the project.",
      "datatype": "choice",
      "choices": [
        {
          "choice": "netcoreapp1.0",
          "description": "Target netcoreapp1.0"
        },
        {
          "choice": "netcoreapp1.1",
          "description": "Target netcoreapp1.1"
        }
      ],
      "defaultValue": "netcoreapp1.1"
    },
    "FrameworkIs1_1": {
      "type": "computed",
      "value": "Framework == \"netcoreapp1.1\""
    },
    "copyrightYear": {
      "type": "generated",
      "generator": "now",
      "replaces": "1975",
      "parameters": {
        "format": "yyyy"
      }
    },
    "NoTools": {
      "type": "parameter",
      "datatype": "bool",
      "defaultValue": "true"
    },
    "skipRestore": {
      "type": "parameter",
      "datatype": "bool",
      "description": "If specified, skips the automatic restore of the project on create.",
      "defaultValue": "false"
    },
    "TargetEditor": {
      "type": "parameter",
      "datatype": "choice",
      "choices": [
        {
          "choice": "dotnetcli",
          "description": "DotNet CLI"
        },
        {
          "choice": "vs2017",
          "description": "Visual Studio 2017"
        }
      ],
      "defaultValue": "dotnetcli",
      "description": "The editor this template is generated for."
    },
    "SeedMethod": {
      "type": "parameter",
      "datatype": "choice",
      "choices": [
        {
          "choice": "OOSeed",
          "description": "Object Orienated seed method"
        },
        {
          "choice": "CSVSeed",
          "description": "seed from a csv file"
        }
      ],
      "description": "Method for seeding the database with data.",
      "defaultValue": "OOSeed"
    }
  },
  "primaryOutputs": [
    {
      "path": "Company.WebApplication1.sln"
    }
  ],
  "defaultName": "WebApplication1",
  "postActions": [
    {
      "condition": "(!skipRestore)",
      "description": "Restore NuGet packages required by this project.",
      "manualInstructions": [
        {
          "text": "Run 'dotnet restore'"
        }
      ],
      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
      "continueOnError": true
    }
  ]
}
```
dotnetcli.host.json
```
{
  "symbolInfo": {
    "TargetFrameworkOverride": {
      "isHidden": "true",
      "longName": "target-framework-override",
      "shortName": ""
    },
    "IncludeApplicationInsights": {
      "isHidden": "true"
    },
    "UseLocalDB": {
      "longName": "use-local-db"
    },
    "AADInstance": {
      "longName": "aad-instance",
      "shortName": "s"
    },
    "OrgReadAccess": {
      "longName": "org-read-access",
      "shortName": "r"
    },
    "ClientId": {
      "longName": "client-id",
      "shortName": ""
    },
    "Domain": {
      "longName": "domain",
      "shortName": ""
    },
    "TenantId": {
      "longName": "tenant-id",
      "shortName": ""
    },
    "Framework": {
      "longName": "framework"
    },
    "NoTools": {
      "isHidden": "true",
      "longName": "no-tools"
    },
    "TargetEditor": {
      "isHidden": "true",
      "longName": "target-editor"
    },
    "skipRestore": {
      "longName": "no-restore",
      "shortName": ""
    },
    "HttpsPort": {
      "isHidden": true
    },
    "KestrelPort": {
      "isHidden": true
    },
    "IISExpressPort": {
      "isHidden": true
    },
    "IncludeLaunchSettings": {
      "isHidden": true
    },
    "UserSecretsId": {
      "isHidden": true
    },
    "SeedMethod": {
      "longName": "seed-method",
      "shortName": ""
    }
  },
  "UsageExamples": [
    "--framework netcoreapp1.1 --auth Individual"
  ]
}
```

**Comment by seancpeters** [comment]
@snebjorn I'm still unable to reproduce the problem against a current new3 build, so I'd say it's most likely the result of an old bug that has since been fixed. 

One thought I had was if those conditionally excluded folders have no files in them in the template source, that would cause them to be always excluded. But as you'd mentioned, with lower-casing, the behavior is as expected, so that shouldn't be the issue.

If they are indeed empty folders, I can explain how to get them to show up.

**Comment by snebjorn** [comment]
Thanks for the quick feedback. I can confirm that it works in the current released CLI (1.0.4) when using `dotnet new` (not new3).
But it seems I broke the new3 install, so I can't confirm if it's working there. But I assume :) So I'm closing the issue.
